lib/prx_auth/rails/ext/controller.rb

```diff
@@ -79,16 +84,33 @@ def lookup_account_names_mapping(ids=current_user.resources)
 
         accounts = URI.open("https://#{id_host}/api/v1/accounts?account_ids=#{ids_param}", options).read
 
-        mapping = JSON.parse(accounts)['accounts'].map { |acct| [acct['id'], acct['display_name']] }.to_h
+        JSON.parse(accounts)['accounts']
+      end
 
-        mapping
+      # token from data set by prx_auth rack middleware
+      def env_token
+        @env_token_data ||= if request.env[PRX_AUTH_ENV_KEY]
+          token_data = request.env[PRX_AUTH_ENV_KEY]
+          PrxAuth::Rails::Token.new(token_data)
```

Comment:
This took me awhile to grok.  But I think what's happening is: PrxAuth rack middleware _may_ have already set the `request.env['prx.auth']` to be a `Rack::PrxAuth::TokenData`.  If the request had an authorization-bearer header.

If not, we fallback to the session-based JWT below.